### PR TITLE
Merge fortuna to master

### PR DESCRIPTION
**Pull request by gorillainduction**
https://www.pivotaltracker.com/story/show/164596030

### CODE REVIEW

File: rebar.lock [hunk 1]
```diff
@@ -34,6 +38,7 @@
   {git,"https://github.com/ninenines/cowlib",
       {ref,"3ef5b48a028bb66f82b452c98ae515903096641c"}},
   1},
+ {<<"eblake2">>,{pkg,<<"eblake2">>,<<"1.0.0">>},1},
```
Comment (lucafavatella) [inline rebar.lock]: https://github.com/aeternity/eblake2/pull/2

File: rebar.lock [hunk 2]
```diff
@@ -16,9 +16,13 @@
...
       {ref,"17d7a8d45894c56aeac08c2f379745b03672e65d"}},
   0},
+ {<<"aeserialization">>,
+  {git,"https://github.com/aeternity/aeserialization.git",
+      {ref,"6dce265753af4e651f77746e77ea125145c85dd3"}},
+  0},
  {<<"aesophia">>,
   {git,"https://github.com/aeternity/aesophia.git",
-      {ref,"4d9d3077ada60baa4ced35a793555888c9684cac"}},
+      {ref,"36395b597a0fa8e9e794f22b4d7caf04fe69fd06"}},
```
[the remark this answers is not in the document]
Comment (lucafavatella) [reply rebar.lock]: Re `{cmd, "cat VERSION | tr -d '[:space:]'"}`
Could you please confirm that deleting the VERSION file in that repo and see if build breaks? I am concerned about `|` losing exit status.

[the remark this answers is not in the document]
Comment (lucafavatella) [reply rebar.lock]: I see compiler version bumped though not included yet in aect_sophia:serialize. Ok.

[the remark this answers is not in the document]
Comment (gorillainduction) [reply rebar.lock]: The build does not break:
```
tobias@Titan:~/GIT/aesophia$ cat _build/default/lib/aesophia/ebin/aesophia.app 
{application,aesophia,
             [{description,"Contract Language for aeternity"},
              {vsn,"cat: VERSION: No such file or directory"},
              {registered,[]},
              {applications,[kernel,stdlib,jsx,syntax_tools,getopt,
                             aebytecode]},
              {env,[]},
              {modules,[aeso_abi,aeso_aci,aeso_ast,aeso_ast_infer_types,
                        aeso_ast_to_icode,aeso_blake2,aeso_builtins,
                        aeso_compiler,aeso_constants,aeso_heap,aeso_icode,
                        aeso_icode_to_asm,aeso_icode_to_fate,aeso_memory,
                        aeso_parse_lib,aeso_parser,aeso_pretty,aeso_scan,
                        aeso_scan_lib,aeso_sophia,aeso_syntax,
                        aeso_syntax_utils,aeso_utils,aesophia]},
              {maintainers,[]},
              {licenses,[]},
              {links,[]}]}.
```

[the remark this answers is not in the document]
Comment (lucafavatella) [reply rebar.lock]: Are the release notes in  aesophia `docs/RELEASE-NOTES-2.0.0.md` meant to go in the node OTP release? It seems to me they would not be included as is.

[the remark this answers is not in the document]
Comment (hanssv) [reply rebar.lock]: Compiler version is included when compiling with `aesophia_http` though...

File: rebar.config [hunk 3]
```diff
@@ -58,10 +58,13 @@
                    {ref, "17d7a8d"}}},
 
         {aebytecode, {git, "https://github.com/aeternity/aebytecode.git",
-                     {ref,"720510a"}}},
+                     {ref,"34ae94e"}}},
 
         {aesophia, {git, "https://github.com/aeternity/aesophia.git",
-                     {ref,"4d9d307"}}},
+                     {ref,"36395b5"}}},
```
Comment (lucafavatella) [inline rebar.config]: It looks to me that we shall put a comment in this line reminding developer that the version dictated by this software may end up in the bytecode/transaction so for the sake of determinism reviewer shall be alerted that any time this line is changed then he/she shall run extra check on version having been adhered to. Not in scope of this PR though.

Comment (hanssv) [reply rebar.config]: Not a problem, this compiler is _NOT_ used to compile :-)

Comment (hanssv) [reply rebar.config]: The compiler used by the (deprecated) HTTP API is the checked in escript...

File: rebar.lock (hunk 2 above)
[the remark this answers is not in the document]
Comment (lucafavatella) [reply rebar.lock]: Rest of the diff  of this dep seems fine (I admit I did not check transitive deps).

[the remark this answers is not in the document]
Comment (hanssv) [reply rebar.lock]: There is no one using the OTP release of `aesophia` and it is not put anywhere...

**Comment by ThomasArts** [comment]
> Requesting changes for tracking breaking build when picking up VERSION in dep fails.

sounds to me like we can do this in a completely new ticket, has little to do with this merge

File: rebar.lock (hunk 2 above)
[the remark this answers is not in the document]
Comment (dincho) [reply rebar.lock]: Perhaps
```
set -o pipefail && cat VERSION | tr -d '[:space:]'
```

[the remark this answers is not in the document]
Comment (lucafavatella) [reply rebar.lock]: @dincho Is pipefail bash specific? Who guarantees that that snippet is run with bash?

[the remark this answers is not in the document]
Comment (gorillainduction) [reply rebar.lock]: I suggest leaving this line as is, but adding the following to rebar.config in aesophia:
```
{pre_hooks, [{"(linux|darwin|solaris|freebsd|netbsd|openbsd)",
              app_compile,
              "test -e VERSION"}
             ]}.
```
I'm not entirely sure this works on windows, though, but I'm sure CI will tell me ;-)

**Comment by lucafavatella** [comment]
I saw [CI system smoke tests job failing on CI](https://circleci.com/gh/aeternity/aeternity/14243):
```
===> Fetching eper ({pkg,<<"eper">>,<<"0.99.1">>})
===> Failed to fetch and copy dep: eper-0.99.1
make: *** [internal-compile-deps] Error 1
Makefile:352: recipe for target 'internal-compile-deps' failed
```
It seems https://hex.pm/packages/eper/0.99.1 temporarily down.

I am taking the liberty to restart CI.

**Comment by lucafavatella** [comment]
eper failure ^^^ happened again in https://circleci.com/gh/aeternity/aeternity/14245

I can reproduce locally running `make system-test-deps`

**Comment by lucafavatella** [comment]
Pushed on distinct branch system-tests https://circleci.com/gh/aeternity/aeternity/14246
Checking whether it is re branch name.

**Comment by lucafavatella** [comment]
And master https://circleci.com/gh/aeternity/aeternity/14242 fails on:
```
===> Package not found in any repo: certifi 2.4.2
make: *** [internal-compile-deps] Error 1
```

**Pull request merged**